src/EditorFeatures/TestUtilities/StubLocalRegistry.cs

```diff
@@ -0,0 +1,105 @@
+
+internal sealed class StubLocalRegistry : ILocalRegistry5, ILocalRegistry4, ILocalRegistry3
+{
+    int ILocalRegistry.CreateInstance(Guid clsid, object punkOuter, ref Guid riid, uint dwFlags, out IntPtr ppvObj)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry.GetTypeLibOfClsid(Guid clsid, out ITypeLib pptLib)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry.GetClassObjectOfClsid(ref Guid clsid, uint dwFlags, IntPtr lpReserved, ref Guid riid, out IntPtr ppvClassObject)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry2.CreateInstance(Guid clsid, object punkOuter, ref Guid riid, uint dwFlags, out IntPtr ppvObj)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry2.GetTypeLibOfClsid(Guid clsid, out ITypeLib pptLib)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry2.GetClassObjectOfClsid(ref Guid clsid, uint dwFlags, IntPtr lpReserved, ref Guid riid, IntPtr ppvClassObject)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry2.GetLocalRegistryRoot(out string pbstrRoot)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.CreateInstance(Guid clsid, object punkOuter, ref Guid riid, uint dwFlags, out IntPtr ppvObj)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.GetTypeLibOfClsid(Guid clsid, out ITypeLib pptLib)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.GetClassObjectOfClsid(ref Guid clsid, uint dwFlags, IntPtr lpReserved, ref Guid riid, IntPtr ppvClassObject)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.GetLocalRegistryRoot(out string pbstrRoot)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.CreateManagedInstance(string codeBase, string assemblyName, string typeName, ref Guid riid, out IntPtr ppvObj)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry3.GetClassObjectOfManagedClass(string codeBase, string assemblyName, string typeName, ref Guid riid, out IntPtr ppvClassObject)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry4.RegisterClassObject(ref Guid rclsid, out uint pdwCookie)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry4.RevokeClassObject(uint dwCookie)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry4.RegisterInterface(ref Guid riid)
+    {
+        throw new NotImplementedException();
+    }
+
+    int ILocalRegistry4.GetLocalRegistryRootEx(uint dwRegType, out uint pdwRegRootHandle, out string pbstrRoot)
+    {
+        pdwRegRootHandle = unchecked((uint)__VsLocalRegistryRootHandle.RegHandle_CurrentUser);
+        pbstrRoot = "Software\\Microsoft\\VisualStudio\\17.0_Test";
```

Comment:
I admit this worries me since something might actually try to use this key and we really don't want that in unit tests.

Reply:
➡️ Me too, but we aren't _supposed_ to be doing this and at least it's constrained to a sane location (similar to allowing test code to write to a subdirectory of %TEMP%).